internal/integrations/rawtemplate_test.go

```diff
@@ -0,0 +1,262 @@
+						},
+					},
+				},
+			},
+		},
+		{
+			title: "complex type in variable",
+			file:  "template-simple.yaml",
+			pipelines: []pipeline.ID{
+				pipeline.MustNewID("metrics"),
+			},
+			params: map[string]any{
+				"somevalue": map[string]any{
+					"subsetting": "foo",
+					"array":      []int{1, 2, 3},
+				},
+			},
+			expected: Config{
+				Receivers: map[component.ID]ComponentConfig{
+					component.MustNewID("foo"): map[string]any{
+						"somesetting": map[string]any{
+							"subsetting": "foo",
+							"array":      []int{1, 2, 3},
+						},
+					},
+				},
+				Processors: map[component.ID]ComponentConfig{
+					component.MustNewID("otherprocessor"): nil,
+					component.MustNewID("someprocessor"):  nil,
+				},
+				Pipelines: map[pipeline.ID]PipelineConfig{
+					pipeline.MustNewID("metrics"): {
+						Receiver: idPtr(component.MustNewID("foo")),
+						Processors: []component.ID{
+							component.MustNewID("someprocessor"),
+							component.MustNewID("otherprocessor"),
+						},
+					},
+				},
+			},
+		},
+		{
+			title: "selected pipeline with name",
+			file:  "template-simple.yaml",
+			pipelines: []pipeline.ID{
+				pipeline.MustNewIDWithName("logs", "raw"),
+			},
+			params: map[string]any{
+				"value":      "bar",
+				"othervalue": "baz",
+			},
+			expected: Config{
+				Receivers: map[component.ID]ComponentConfig{
+					component.MustNewID("bar"): map[string]any{
+						"somecomplexsetting": map[string]any{
+							"someobject": map[string]any{
+								"value": "bar",
+								"other": "baz",
+							},
+						},
+					},
+				},
+				Processors: nil,
+				Pipelines: map[pipeline.ID]PipelineConfig{
+					pipeline.MustNewIDWithName("logs", "raw"): {
+						Receiver:   idPtr(component.MustNewID("bar")),
+						Processors: []component.ID{},
+					},
+				},
+			},
+		},
+		{
+			title: "missing variable",
+			file:  "template-simple.yaml",
+			pipelines: []pipeline.ID{
+				pipeline.MustNewID("metrics"),
+			},
+			expectedErr: "variable \"somevalue\" not found",
+		},
+		{
+			title: "missing pipeline",
+			file:  "template-simple.yaml",
+			pipelines: []pipeline.ID{
+				pipeline.MustNewID("traces"),
+			},
+			expectedErr: "selecting pipelines: component \"traces\" not found",
+		},
+		{
+			title:   "missing receiver",
+			file:    "template-missing-receiver.yaml",
+			fileErr: "receiver \"foo/missing\" not defined",
+		},
+		{
+			title:   "missing processor",
+			file:    "template-missing-processor.yaml",
+			fileErr: "processor \"third/missing\" not defined",
+		},
+		{
+			title:   "missing processor",
+			file:    "template-unknown-fields.yaml",
+			fileErr: "field extensions not found",
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.title, func(t *testing.T) {
+			var template Integration
+			var err error
+			template, err = NewRawTemplate(readTemplateFile(c.file))
+			if c.fileErr != "" {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), c.fileErr)
```

Comment:
You can use [assert.ErrorContains](https://pkg.go.dev/github.com/stretchr/testify/assert#ErrorContains) instead if you prefer:

```suggestion
				assert.ErrorContains(t, err, c.fileErr)
```

Reply:
Nice one, changed, thanks.